src/Commands/SvnTag.php

```diff
@@ -0,0 +1,597 @@
+
+			$output->writeln( '' );
+
+			// Remove deleted files
+			$output->writeln( "Deleting all removed files from destination Tag folder:" );
+			foreach ( $scan['deleted'] as $file ) {
+				$remove_cmd = "svn rm tags/{$destination_tag}/{$file}";
+				$output->writeln( $remove_cmd );
+				shell_exec( $cd_to_repo_cmd . $remove_cmd );
+			}
+		} else {
+			$output->writeln( "No files removed from '{$source_tag}' to '{$destination_tag}'." );
+		}
+
+		$output->writeln( '' );
+
+		// Commit changes to SVN.
+		$commit_cmd = "svn commit tags/{$destination_tag} -m 'Apply modifications to {$destination_tag}'";
+		$output->writeln( "Executing SVN Command:  \n" . $commit_cmd );
+		shell_exec( $cd_to_repo_cmd . $commit_cmd );
+
+		/**
+		 * The check sum below doesnt work, need some more work before its ready.
+		 *
+		 * $svn_checksum = shell_exec( "svn info --show-item=checksum {$svn_url}/tags/{$destination_tag}" );
+		 * $zip_checksum = sha1_file( $zip_file );
+		 *
+		 * if ($svn_checksum !== $zip_checksum) {
+		 * $output->writeln('<error>Checksum mismatch.</error>');
+		 * return self::CMD_FAILURE;
+		 * }
+		 */
+
+		return $this->cleanup_to_success( "Operation completed successfully." );
+	}
+
+	/**
+	 * Compare two directories and return an array of added and deleted files.
+	 *
+	 * @since 1.2.8
+	 *
+	 * @param string $source      Directory A.
+	 * @param string $destination Directory B.
+	 *
+	 * @return array|null
+	 */
+	protected function compare_directories( $source, $destination ) {
+		// Ensure directories end with '/'
+		$source      = rtrim( $source, '/' ) . '/';
+		$destination = rtrim( $destination, '/' ) . '/';
+
+		// Check if directories exist
+		if ( ! is_dir( $source ) || ! is_dir( $destination ) ) {
+			return null;
+		}
+
+		// Execute rsync command to check for added files.
+		$command = "rsync --dry-run --itemize-changes --recursive --ignore-existing {$destination}/ {$source}/ | grep '>f+' | awk '{print \$2}'";
+		exec( $command, $added_in_source );
+
+		// Execute rsync command to check for deleted files.
+		$command = "rsync --dry-run --itemize-changes --recursive --ignore-existing {$source}/ {$destination}/ | grep '>f+' | awk '{print \$2}'";
+		exec( $command, $removed_from_source );
+
+		return [
+			'deleted' => array_values( $removed_from_source ),  // Re-index the array
+			'added' => array_values( $added_in_source ),    // Re-index the array
+		];
+	}
+
+	/**
+	 * Cleanup the workspace and return error.
+	 *
+	 * @since 1.2.8
+	 *
+	 * @param string|null $message
+	 *
+	 * @return int
+	 */
+	protected function cleanup_to_error( string $message = null ): int {
+
+		$this->cleanup();
+
+		$this->get_output()->writeln( $message );
+		return self::CMD_FAILURE;
+	}
+
+	/**
+	 * Cleanup the workspace and return success.
+	 *
+	 * @since 1.2.8
+	 *
+	 * @param string|null $message
+	 *
+	 * @return int
+	 */
+	protected function cleanup_to_success( string $message = null ): int {
```

Comment:
have to admit this feels les D.R.Y. than a single function that takes second arg

Reply:
Idea is to avoid confusion, but I could make another method behind this one like `cleanup_to( $status, $message)` but felt over kill for now.